src/platform/nrfconnect/wifi/ConnectivityManagerImplWiFi.cpp

```diff
@@ -39,124 +39,200 @@ using namespace ::chip::TLV;
+            : ConnectivityManager::WiFiStationMode::kWiFiStationMode_Enabled;
+    }
+    return mStationMode;
+}
+
+CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiStationMode(ConnectivityManager::WiFiStationMode aMode)
+{
+    VerifyOrReturnError(ConnectivityManager::WiFiStationMode::kWiFiStationMode_NotSupported != aMode,
+                        CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+
+    if (aMode != mStationMode)
+    {
+        mStationMode = aMode;
+        if (mStationMode != ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled)
+        {
+            bool doEnable{ ConnectivityManager::WiFiStationMode::kWiFiStationMode_Enabled == mStationMode };
+            if (doEnable)
+            {
+                OnStationConnected();
+            }
+            else
+            {
+                WiFiManager().Instance().DisconnectStation();
+                OnStationDisconnected();
+            }
+        }
+    }
+
     return CHIP_NO_ERROR;
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationEnabled(void)
 {
-    return false;
+    return (WiFiManager::StationStatus::DISABLED <= WiFiManager().Instance().GetStationStatus());
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationApplicationControlled(void)
 {
-    return false;
+    return (ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled == mStationMode);
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationConnected(void)
 {
-    return false;
+    return (WiFiManager::StationStatus::FULLY_PROVISIONED == WiFiManager().Instance().GetStationStatus());
 }
 
 System::Clock::Timeout ConnectivityManagerImplWiFi::_GetWiFiStationReconnectInterval(void)
 {
-    return System::Clock::kZero;
+    return mWiFiStationReconnectInterval;
 }
 
 CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiStationReconnectInterval(System::Clock::Timeout val)
 {
+    mWiFiStationReconnectInterval = val;
     return CHIP_NO_ERROR;
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationProvisioned(void)
 {
-    return false;
+    return (WiFiManager::StationStatus::CONNECTED <= WiFiManager().Instance().GetStationStatus());
 }
 
-void ConnectivityManagerImplWiFi::_ClearWiFiStationProvision(void) {}
-
-ConnectivityManager::WiFiAPMode ConnectivityManagerImplWiFi::_GetWiFiAPMode(void)
+void ConnectivityManagerImplWiFi::_ClearWiFiStationProvision(void)
 {
-    return ConnectivityManager::WiFiAPMode::kWiFiAPMode_Disabled;
+    if (_IsWiFiStationProvisioned())
+    {
+        if (CHIP_NO_ERROR != WiFiManager().Instance().ClearStationProvisioningData())
+        {
+            ChipLogError(DeviceLayer, "Cannot clear WiFi station provisioning data");
+        }
+    }
 }
 
-CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiAPMode(ConnectivityManager::WiFiAPMode val)
+bool ConnectivityManagerImplWiFi::_CanStartWiFiScan()
 {
-    return CHIP_NO_ERROR;
+    return (WiFiManager::StationStatus::CONNECTING < WiFiManager().Instance().GetStationStatus());
```

Comment:
Hmm. Can't we do it not being connected? I think we can.

Reply:
So maybe let's reconsider when actually the scan is not possible. I would say when scanning already in progress, when the station is disabled (WPA_INTERFACE_DISABLED, WPA_INACTIVE) and maybe when connecting?

Reply:
Agree.